Sources/Core/Client/Client+RxConnection.swift

```diff
@@ -7,70 +7,40 @@
 //
 
 import UIKit
+import StreamChatClient
 import RxSwift
 
-// MARK: Connection
+// MARK: RxConnection
 
-extension Reactive where Base == Client {
+extension Client {
+    private static var rxConnectionKey: UInt8 = 0
     
-    /// An observable connection.
-    public var connection: Observable<WebSocket.Connection> {
-        return base.rxConnection
-    }
-    
-    func createConnection() -> Observable<WebSocket.Connection> {
-        if let token = base.token, let error = base.checkUserAndToken(token) {
-            return .error(error)
-        }
-        
-        let app = UIApplication.shared
-        
-        let appState = isTestsEnvironment()
-            ? .just(.active)
-            : app.rx.appState
-                .filter { $0 != .inactive }
-                .distinctUntilChanged()
-                .startWith(app.appState)
-                .do(onNext: { state in
-                    if Client.shared.logOptions.isEnabled {
-                        ClientLogger.log("📱", "App state \(state)")
-                    }
+    fileprivate var rxConnection: Observable<Connection> {
```

Comment:
Why do we have `rxConnection` and `rx.connection`, pointing to same logic but having 2 names? I propose we get rid of `rxConnection` and move logic under `rx.connection`

Reply:
`rxConnection` is an internal variable. It holds a shared instance of Observable.
`Client.shared.rx.connection` is a public Observable on the rx subset.

Reply:
Can't we solve this issue without having 2 properties?

Reply:
Nope. We have to store the Observable instance and only in the class-based type.
The `rx.connection` is a read-only property.

Reply:
Can't we keep `private static var rxConnectionKey: UInt8 = 0` inside the class extension but move the actual logic under `rx.connection`?

Reply:
It's already `private` and `fileprivate` implementation. It was in rx subset but generates a useless code. Now it's pretty clear and in the right place.

Reply:
I understand that you worried about extension properties, but we have no choice with moving RxSwift out from low-level Client. I tried to simplify and clear how to add rx properties. They are all aligned with the same pattern.

Reply:
I was worried about having 2 separate properties and one pointing to the other.
`rx.connection` -> `rxConnection` -> actual logic
I'm suggesting eliminating the unnecessary `rxConnection`, not for users (because they can't see it - it's `fileprivate`) but for maintainers.

Reply:
You don't understand me. You have to store logic in the base.

Reply:
Yes, there's a communication issue. I don't understand why we need `rxConnection`.
I've tried and I could eliminate the `rxConnection` intermediate variable with this code:
```swift
// MARK: RxConnection

extension Client {
    fileprivate static var rxConnectionKey: UInt8 = 0
}

// MARK: - Connection

extension Reactive where Base == Client {
    /// An observable connection.
    public var connection: Observable<Connection> {
        associated(to: base, key: &Client.rxConnectionKey) { [unowned base] in
            let connection = onConnect.filter { _ in !base.isExpiredTokenInProgress }
            let appState = UIApplication.shared.rx.applicationState.filter({ $0 != .inactive })
            let isInternetAvailable: Observable<Bool> = InternetConnection.shared.rx.isAvailable
            
            let environment = Observable.combineLatest(appState, isInternetAvailable)
                .distinctUntilChanged({ $0.0 == $1.0 && $0.1 == $1.1 })
                .observeOn(MainScheduler.instance)
                .do(onNext: { appState, isInternetAvailable in
                    base.logger?.log("Environment: appState: \(appState), Internet: \(isInternetAvailable ? "yes" : "no")")
                    base.handleConnection(appState: appState, isInternetAvailable: isInternetAvailable)
                })
            
            return Observable.combineLatest(connection, environment)
                .map { connection, _ in connection }
                .distinctUntilChanged()
                .do(onDispose: { base.disconnect() })
                .share(replay: 1)
        }
    }
}
```
Is there something I'm missing?

Reply:
This is perfect! I missed we can associate with `base`. I'll do the update.